exchanges/orderbook/orderbook.go

```diff
@@ -13,6 +13,39 @@ import (
 	"github.com/thrasher-corp/gocryptotrader/log"
 )
 
+// GetAveragePrice calculates the average price for a buy of a specified amount
+func (b *Base) GetAveragePrice(buy bool, amount float64) (float64, error) {
+	remainingAmount := amount
+	var aggNominalAmount float64
+	if buy {
+		for x := range b.Asks {
+			if remainingAmount <= b.Asks[x].Amount {
+				aggNominalAmount += b.Asks[x].Price * remainingAmount
+				remainingAmount = 0
+				break
+			} else {
+				aggNominalAmount += b.Asks[x].Price * b.Asks[x].Amount
+				remainingAmount -= b.Asks[x].Amount
+			}
+		}
```

Comment:
create a function for this and below they seem pretty similar in nature and just pass by reference the required variables so it can inline.